src/test/java/org/openrewrite/java/testing/junit5/AddParameterizedTestAnnotationTest.java

```diff
@@ -0,0 +1,116 @@
+package org.openrewrite.java.testing.junit5;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.Issue;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+public class AddParameterizedTestAnnotationTest implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec
+          .parser(JavaParser.fromJavaVersion()
+            .classpathFromResources(new InMemoryExecutionContext(), "junit-4.13", "junit-jupiter-api-5.9"))
+          .recipe(new AddParameterizedTestAnnotation());
+    }
+
+    @Issue("https://github.com/openrewrite/rewrite-testing-frameworks/issues/314")
+    @Test
+    void replaceTestWithParameterizedTest() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.junit.jupiter.api.Test;
+              import org.junit.jupiter.params.provider.ValueSource;
+              
+              class NumbersTest {
+                @Test
+                @ValueSource(ints = {1, 3, 5, -3, 15,Integer.MAX_VALUE})
+                void testIsOdd(int number) {
+                    assertTrue(number % 2 != 0);
+                }
+              }
+              """,
+            """
+              import org.junit.jupiter.params.ParameterizedTest;
+              import org.junit.jupiter.params.provider.ValueSource;
+              
+              class NumbersTest {
+                @ParameterizedTest
+                @ValueSource(ints = {1, 3, 5, -3, 15,Integer.MAX_VALUE})
+                void testIsOdd(int number) {
+                    assertTrue(number % 2 != 0);
+                }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void onlyReplacesWithValueSourceAnnotation() {
+        /*
+        This test ensures that the recipe will only run on code that includes a
+        @ValueSource(...) annotation.
+         */
+        rewriteRun(
+          java(
+            """
+              @Test
```

Comment:
This test is lacking imports, and thus likely to have missing types, and through that failing matchers.

Same goes for the tests below.

Reply:
It might help to check that the code snippets that you use in a text block are actually valid stand alone classes, by briefly copy-pasting these into a new file. Your IDE will then immediately tell you if you're missing imports for instance.